hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestAlterTable.scala

```diff
@@ -200,6 +201,13 @@ class TestAlterTable extends HoodieSparkSqlTestBase {
         checkAnswer(s"select id, name, price, ts, dt from $tableName2")(
           Seq(1, "a1", 10.0, 1000, null)
         )
+
+        if (HoodieSparkUtils.gteqSpark3_1) {
+          withSQLConf("hoodie.schema.on.read.enable" -> "true") {
```

Comment:
Can you explain a little more why we need this?

Reply:
AlterTableCommand only work on spark datasourcev2 which is controlled by `hoodie.schema.on.read.enable`